uber/site_sections/budget.py

```diff
@@ -46,3 +46,157 @@ def mpoints(self, session):
         all = [(sum(mpu.amount for mpu in mpus), group, mpus)
                for group, mpus in groups.items()]
         return {'all': sorted(all, reverse=True)}
+
+    @ajax
+    def add_promo_code_words(self, session, text='', part_of_speech=0):
```

Comment:
I hate to point this out after you've already done all the work, but is the half-ton of extra scaffolding around promo code words really worth it? Do we know if they will actually be useful to admins, or should we consider adding them via plugin instead? I know I said I wanted to keep them, but they're like their own extra feature on top of promo codes themselves.